### PR TITLE
[Core] Indent write events in PrettyFormatter 

### PR DESCRIPTION
**Pull request by alexandreonterroso**
## Summary

Add indentation to PrettyFormatter

## Motivation and Context

As @mpkorstanje said in #1784 , scenario.write() ignores the steps indentation so it doesn't look so pretty.

### CODE REVIEW

File: core/src/main/java/cucumber/runtime/formatter/PrettyFormatter.java [hunk 1]
```diff
@@ -150,7 +151,9 @@ private void handleTestStepFinished(TestStepFinished event) {
     }
 
     private void handleWrite(WriteEvent event) {
-        out.println(event.text);
+        out.println();
+        out.println(STEP_SCENARIO_INDENT + event.text);
+        out.println();
```
Comment (mpkorstanje) [inline core/src/main/java/cucumber/runtime/formatter/PrettyFormatter.java]: You may want to test this with multi-line strings.

Comment (alexandreonterroso) [reply core/src/main/java/cucumber/runtime/formatter/PrettyFormatter.java]: You are right, I added some changes

File: core/src/main/java/cucumber/runtime/formatter/PrettyFormatter.java [hunk 2]
```diff
@@ -150,8 +151,18 @@ private void handleTestStepFinished(TestStepFinished event) {
...
-        out.println(event.text);
-    }
+        out.println();
+        try(BufferedReader lines = new BufferedReader(new StringReader(event.text))) {
+		String line = lines.readLine();
+		while(line != null) {
+			out.println(STEP_SCENARIO_INDENT + line);
+			line = lines.readLine();
+		}
+	} catch (IOException e) {
```
Comment (mpkorstanje) [inline core/src/main/java/cucumber/runtime/formatter/PrettyFormatter.java]: The canonical way to write this avoids the repetition of `readLine`. It is a little bit less error prone.

```
            String line;
            while ((line = lines.readLine()) != null) {
                out.println(STEP_SCENARIO_INDENT + line);
            }
```

Aside from that looks good. Don't forget to indent your code properly.

File: core/src/main/java/cucumber/runtime/formatter/PrettyFormatter.java [hunk 3]
```diff
@@ -150,8 +151,17 @@ private void handleTestStepFinished(TestStepFinished event) {
     }
 
     private void handleWrite(WriteEvent event) {
-        out.println(event.text);
-    }
+        out.println();
+        try(BufferedReader lines = new BufferedReader(new StringReader(event.text))) {
+		String line;
```
Comment (mpkorstanje) [inline core/src/main/java/cucumber/runtime/formatter/PrettyFormatter.java]: Looks better. But still not quite right. If you are using an IDE there usually is a short cut to do this for you.

File: core/src/main/java/cucumber/runtime/formatter/PrettyFormatter.java [hunk 4]
```diff
@@ -150,8 +151,17 @@ private void handleTestStepFinished(TestStepFinished event) {
     }
 
     private void handleWrite(WriteEvent event) {
-        out.println(event.text);
-    }
+        out.println();
+        try(BufferedReader lines = new BufferedReader(new StringReader(event.text))) {
+			String line;
+			while ((line = lines.readLine()) != null) {
```
Comment (mpkorstanje) [inline core/src/main/java/cucumber/runtime/formatter/PrettyFormatter.java]: I don't mean to keep going on about this but please use spaces for indentation rather then tabs.

We've documented the settings to configure your IDE here:

https://github.com/cucumber/cucumber-jvm/blob/master/CONTRIBUTING.md#contributing

And we also provide a .editorconfig file that most IDEs will pick up automatically.

https://github.com/cucumber/cucumber-jvm/blob/master/.editorconfig

**Comment by mpkorstanje** [comment]
Had to manually merge this because your repo was lagging behind Cucumber JVM quite a bit. 

Cheers!

**Pull request closed without merging**